client/state/comments/reducer.js

```diff
@@ -113,14 +114,20 @@ export function items( state = {}, action ) {
  * @returns {Object} new redux state
  */
 export function totalCommentsCount( state = {}, action ) {
+	const { siteId, postId } = action;
+	const stateKey = getStateKey( siteId, postId );
+
 	switch ( action.type ) {
 		case COMMENTS_COUNT_RECEIVE:
-			const { siteId, postId } = action;
-			const stateKey = getStateKey( siteId, postId );
 			return {
 				...state,
 				[ stateKey ]: action.totalCommentsCount
 			};
+		case COMMENTS_COUNT_INCREMENT:
+			return {
+				...state,
+				[ stateKey ]: state[ stateKey ] + 1
+			};
```

Comment:
If we're adding a new comment with the new `COMMENT_WRITE` action there's no reason to create this new action. Reducers are made for listening to disparate events in order to update their own data.

Could this work too?
```js
case COMMENTS_WRITE:
	return {
		...state,
		[ stateKey ]: state[ stateKey ] + 1
	};
```